### PR TITLE
Added URL decode filter to common for import into projects that need it

### PR DESCRIPTION
**Pull request by harageth**
# Resolves

https://jira.rax.io/browse/SALUS-892

# What

Add a Filter that will URLDecode the path coming in so that we have the proper values.

# How

We URLDecode the path as it comes in and then inject it back into the request for deserialization into the parameters in the APIControllers

# How to test

We currently aren't testing the other Filters that we have so I didn't add unit tests. I am not sure if it makes sense to add unit tests for the filter in Common either. 

# Why

This is much simpler than adding a URLDecode on every parameter that we receive in every APIController

# TODO

Add the @Configuration files in each of the projects that need to utilize this

### CODE REVIEW

File: src/main/java/com/rackspace/salus/common/web/URLDecodeFilter.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+public class URLDecodeFilter implements Filter {
+
+  @Override
+  public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+      FilterChain filterChain) throws IOException, ServletException {
+
+    HttpServletRequest req = (HttpServletRequest) servletRequest;
+    req.setAttribute(
+        WebUtils.INCLUDE_SERVLET_PATH_ATTRIBUTE, URLDecoder.decode(req.getRequestURI(), "UTF-8"));
```
Comment (jjbuchan) [inline src/main/java/com/rackspace/salus/common/web/URLDecodeFilter.java:41]: > "UTF-8"

Is there a static variable for this?  Something that already exists in the java libs?

File: src/main/java/com/rackspace/salus/common/web/URLDecodeFilter.java [hunk 2]
```diff
@@ -26,9 +27,8 @@
 /**
  * A filter to make sure that our inputs are properly URLdecode'ed
  *
- * To apply this to a particular project a configuration needs to be written
- * to apply this to specific URL patterns. It is likely ok to apply this to
- * every URI in a project.
+ * To apply this to a project add @EnableURLDecode to the main function that starts spring boot
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/common/web/URLDecodeFilter.java:30]: `@` symbols are special in Javadoc, so replacing with a link is an easy solution:
```suggestion
 * To apply this to a project add {@link EnableURLDecode} to the main function that starts spring boot
```

**Comment by harageth** [comment]
I suppose this bears the question of whether we want to still add this filter since I actually fixed the client? 

I know @jjbuchan is probably against it and that's legitimate. It would be an extra security for us making sure it doesn't happen again in the future.

**Comment by jjbuchan** [comment]
Yeah, I'm good with closing this even though it has the potential to save us from hitting problems later.  But generally I like knowing there is a problem to solve vs. it potentially being masked.  

If we decide we don't care about a problem enough / don't want to fix it a different way then I'd say we can come back to this.

**Comment by itzg** [comment]
I agree with @jjbuchan

**Pull request closed without merging**